### PR TITLE
Use intersphinx for magicgui objects in `magicgui.md`

### PR DESCRIPTION
**Pull request by lucyleeow**
Some links were broken so I have amended all to use magicgui intersphinx instead. I think the links were there in the original document, not sure why intersphinx wasn't used.

### CODE REVIEW

**Review by psobolewskiPhD — APPROVED** [review]

File: docs/howtos/extending/magicgui.md [hunk 1]
```diff
@@ -102,17 +101,13 @@ widget_demo.show()
...
 parameters. In the example above, `call_button` specifies the button text and the
 `param_options` `slider_float` and `dropdown` let you customize the widget
 associated with those function parameters.
 
 Third party packages (like napari in this case) may provide support for their types
-using
-[`magicgui.register_type`](https://pyapp-kit.github.io/magicgui/usage/types_widgets.html#register-type).
-Indeed napari uses
-[`magicgui.register_type`](https://pyapp-kit.github.io/magicgui/usage/types_widgets.html#register-type)
+using {func}`magicgui.register_type`. Indeed napari uses {func}`magicgui.register_type`
```
Comment (melissawm) [inline docs/howtos/extending/magicgui.md]: These links seem to not be resolving, I think we need
```suggestion
using {func}`magicgui.type_map.register_type`. Indeed napari uses {func}`magicgui.type_map.register_type`
```

**Review by melissawm — APPROVED** [review]
Small note, other than that looks good, thanks @lucyleeow !

File: docs/howtos/extending/magicgui.md (hunk 1 above)
Comment (lucyleeow) [reply docs/howtos/extending/magicgui.md]: Thanks, sorry I forgot to check the links. I was trying to use the same object path as I saw in magicgui docs (see [here](https://github.com/pyapp-kit/magicgui/blob/1e78e6f0bee37aeb9ce7d300a52f5f120141f834/docs/type_map.md?plain=1#L273)) but Talley must use different API auto doc thing for magicgui.

Comment (lucyleeow) [reply docs/howtos/extending/magicgui.md]: That fixed it, I wonder what Talley uses internally.

**Pull request merged**